signer/src/stacks/api.rs

```diff
@@ -902,11 +896,23 @@ impl StacksInteract for StacksClient {
     where
         T: AsTxPayload + Send + Sync,
     {
+        // Consensus serialize the transaction payload to bytes. This is the
+        // method that the stacks node uses to determine transaction size when
+        // verifying admittance into the mempool.
+        let mut payload_bytes = Vec::new();
+        payload
+            .tx_payload()
+            .consensus_serialize(&mut payload_bytes)
```

Comment:
Oh, I think we'll need to modify this slightly. Since `tx_payload()` returns a `TransactionPayload` object, it is missing things that we need in the transaction, like version, chain id, signatures, and a few other things. These things increase the size of the transaction and distort the fee.

I think we should consider adding a function that creates a "dummy" transaction from the payload and the multi-sig wallet parameters that will have the exact size as the "proper" transaction but with invalid signatures.

Reply:
Yeah I wasn't sure if it's like Bitcoin or not where witness data is treated differently etc. It was problematic I noticed to do this both pre-signing and post-signing since pre-signing we don't have all the signatures, and post-signing the fee may change.

I'll check out the dummy transaction approach.